### PR TITLE
Add building highlight on arrival

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

Resolves https://github.com/mapbox/navigation-sdks/issues/819

[Android-iOS product spec](https://docs.google.com/document/d/16yRtY_U-0fGSedg1heEu4TaIWgTnNov4HZX84bG270U/edit?usp=sharing)
[Android data transfer spec](https://docs.google.com/document/d/1OdcPyUPTWvjS5ucH7Js9bbnmt91dRv9injl7DNCS03s/edit)

Creating a draft as an example. Opening this so it's easier to discuss what we want for this experience. Then we can look into moving things to the ui sdk.

- [x] Show the building on map click
- [x] Make the building slightly transparent
- [x] Connect the building height to actual map data
- [x] Make an example where all buildings are 3d 
- [x] Hook it up to an arrival experience
- [x] Move into the SDK and define ArrivalOptions

Other issues
 - Using Expressions with Feature ids https://github.com/mapbox/mapbox-maps-android/issues/271
 - FillExtrusionLayer is inside out https://github.com/mapbox/mapbox-maps-android/issues/142
 - How to extract building color from map https://github.com/mapbox/mapbox-maps-android/issues/235
 - Using conditional Expressions https://github.com/mapbox/mapbox-gl-native-internal/issues/1637

### Changelog

```
<changelog>Add building highlight on arrival</changelog>
```

### Screenshots or Gifs

output

cc: @abhishek1508 @cafesilencio

### CODE REVIEW

**Comment by abhishek1508** [comment]
@kmadsen The PR looks good in general. 

One of the important aspect of UI 2.0 is that the UI components will not be responsible to maintain state and will be unbiased on when to make appropriate API calls.
Keeping above in mind, I would recommend the following changes:

- [ ] Looking at `BuildingArrivalApi`, it seems like a Utility API. We are not supporting these kind of API's anymore and let developers decide when to make appropriate calls. So basically, whatever you are doing in `onArrival` observer is what we think should happen, but some other developer might have some other preference. This leads to open question and customization requests. **Remove Building Arrival API**
- [ ] Add a new activity `MapboxBuildingHighlightActivity` and demonstrate the use case.
- [ ] Change the name from `BuildingHighlightApi` to `MapboxBuildingHighlightApi` to maintain consistency.
- [ ] Add the `BuildingHighlightApi` to a package `api` and `BuildingHighlightOptions` to a package `model` to maintain consistency with all the other features.

**Comment by kmadsen** [comment]
> Looking at BuildingArrivalApi, it seems like a Utility API. ..  Remove Building Arrival API

yeah let's meet, cause I'm confused what our definition of a utility API is (it seems like anything can be defined/not-defined this way)

> Add a new activity MapboxBuildingHighlightActivity and demonstrate the use case.

I'm a bit confused with the changes with `examples` and `qa-test-app`. It is an entire copy paste of `MapboxNavigationActivity` with a couple lines of change.

> Change the name from BuildingHighlightApi to MapboxBuildingHighlightApi to maintain consistency.
> Add the BuildingHighlightApi to a package api and BuildingHighlightOptions to a package model to maintain consistency with all the other features.

👍

File: examples/src/main/java/com/mapbox/navigation/examples/core/MapboxBuildingHighlightActivity.kt [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+     * allows you to select buildings. When running this example, try tapping on a building
+     * to see it highlight.
+     *
+     * The MapboxBuildingArrivalApi will automatically highlight a building when the navigation
+     * has arrived.
+     */
+    private val buildingsArrivalApi = MapboxBuildingArrivalApi()
+    private lateinit var buildingHighlightApi: MapboxBuildingHighlightApi
+
+    private val tripProgressFormatter: TripProgressUpdateFormatter by lazy {
```
Comment (abhishek1508) [inline examples/src/main/java/com/mapbox/navigation/examples/core/MapboxBuildingHighlightActivity.kt:92]: Can we remove `TripProgress` related code?

File: examples/src/main/java/com/mapbox/navigation/examples/core/MapboxBuildingHighlightActivity.kt [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+                    // no impl
+                }
+            }
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = LayoutActivityTripprogressBinding.inflate(layoutInflater)
```
Comment (abhishek1508) [inline examples/src/main/java/com/mapbox/navigation/examples/core/MapboxBuildingHighlightActivity.kt:291]: Let's add a new layout file for this activity
`layout_activity_building_highlight` and then use `LayoutActivityBuildingHighlightBinding`

**Review by abhishek1508 — APPROVED** [review]

**Pull request merged**